ros2_control_demo_hardware/src/rrbot_system_multi_interface.cpp

```diff
@@ -238,7 +240,7 @@ return_type RRBotSystemMultiInterfaceHardware::start()
 
   RCLCPP_INFO(
     rclcpp::get_logger("RRBotSystemMultiInterfaceHardware"), "System successfully started! %u",
-    control_lvl_[0]);
+    static_cast<unsigned int>(control_lvl_[0]));
```

Comment:
are you sure that's not of size type (`size_t`)?

Reply:
The type of control_lvl_[0] is uint8_t:

`  enum class integration_lvl_t : std::uint8_t
  {
    UNDEFINED = 0,
    POSITION = 1,
    VELOCITY = 2,
    ACCELERATION = 3
  };

  std::vector<integration_lvl_t> control_lvl_;
`
Originally I was extracting the underlaying enum type and using it in the static cast, but @bmagyar objected to the complexity.  Not sure how to resolve this?

Reply:
I can see that getting the underlying type might be a bit verbose. In this case though I would cast to the appropriate type which is `std::uint8_t` to be precise on all platforms.

Reply:
Done